siatest/renter/renter_test.go

```diff
@@ -2175,91 +2495,86 @@ func checkContractVsReportedSpending(r *siatest.TestNode, WindowSize types.Block
 			`, fm.DownloadSpending.HumanString(), spending.DownloadSpending.HumanString())
-		return errors.New(err)
 	}
 	// Compare Storage Spending
 	if fm.StorageSpending.Cmp(spending.StorageSpending) != 0 {
-		err := fmt.Sprintf(`Storage spending not equal:
+		return fmt.Errorf(`Storage spending not equal:
 			Financial Metrics SS: %v
 			Contract SS:          %v
 			`, fm.StorageSpending.HumanString(), spending.StorageSpending.HumanString())
-		return errors.New(err)
 	}
 	// Compare Withheld Funds
 	if fm.WithheldFunds.Cmp(spending.WithheldFunds) != 0 {
-		err := fmt.Sprintf(`Withheld Funds not equal:
+		return fmt.Errorf(`Withheld Funds not equal:
 			Financial Metrics WF: %v
 			Contract WF:          %v
 			`, fm.WithheldFunds.HumanString(), spending.WithheldFunds.HumanString())
-		return errors.New(err)
 	}
 	// Compare Release Block
 	if fm.ReleaseBlock != spending.ReleaseBlock {
-		err := fmt.Sprintf(`Release Block not equal:
+		return fmt.Errorf(`Release Block not equal:
 			Financial Metrics RB: %v
 			Contract RB:          %v
 			`, fm.ReleaseBlock, spending.ReleaseBlock)
-		return errors.New(err)
 	}
 	// Compare Previous Spending
 	if fm.PreviousSpending.Cmp(spending.PreviousSpending) != 0 {
-		err := fmt.Sprintf(`Previous spending not equal:
+		return fmt.Errorf(`Previous spending not equal:
 			Financial Metrics PS: %v
 			Contract PS:          %v
 			`, fm.PreviousSpending.HumanString(), spending.PreviousSpending.HumanString())
-		return errors.New(err)
 	}
 
 	return nil
 }
 
 // checkRenewedContracts confirms that renewed contracts have zero upload and
-// download spending
+// download spending. Renewed contracts should be the renter's active contracts
 func checkRenewedContracts(renewedContracts []api.RenterContract) error {
 	for _, c := range renewedContracts {
-		if c.GoodForRenew {
-			if c.UploadSpending.Cmp(types.ZeroCurrency) != 0 && c.GoodForUpload {
-				err := fmt.Sprintf("Upload spending on renewed contract equal to %v, expected zero", c.UploadSpending.HumanString())
-				return errors.New(err)
-			}
-			if c.DownloadSpending.Cmp(types.ZeroCurrency) != 0 {
-				err := fmt.Sprintf("Download spending on renewed contract equal to %v, expected zero", c.DownloadSpending.HumanString())
-				return errors.New(err)
-			}
+		if c.UploadSpending.Cmp(types.ZeroCurrency) != 0 && c.GoodForUpload {
+			return fmt.Errorf("Upload spending on renewed contract equal to %v, expected zero", c.UploadSpending.HumanString())
+		}
+		if c.DownloadSpending.Cmp(types.ZeroCurrency) != 0 {
+			return fmt.Errorf("Download spending on renewed contract equal to %v, expected zero", c.DownloadSpending.HumanString())
 		}
 	}
 	return nil
 }
 
 // renewContractByRenewWindow mines blocks to force contract renewal
+//
+// TODO:
+// 1) remove excess tg.Sync() calls in the test since it is called in this
+// function
+// 2) make this more generic by referencing the contracts
+// 3) look for other places where we are mining blocks in tests and see if
```

Comment:
should probably turn this TODO into a `programming task`

Reply:
Yes I agree, I'll pull out my todo's into a programming task.  I was just documenting them as I went as to not forget.